### ISSUE TITLE
Does not work with Write Defaults off?

### ISSUE THREAD

**Issue by wickles** [theepicsnail/Marker#13]
I tried using the Marker on an AV3 avatar with Write Defaults off and it completely broke everything. Gestures, toggles, and the marker itself became practically nonfunctional. I tried changing the marker behaviors to all have Write Defaults off as well but it did not improve anything.

**Comment by chibiskuld** [comment]
Nope, and not recommended. Write defaults is okay if used properly. The marker is a case where they are. You can have the rest of your controller with it off, but do not turn it off for the marker layers.

**Comment by wickles** [comment]
If I leave Write Defaults on for the marker layers and Write Defaults off for the other layers, then gesture expressions and such seem to be okay and the marker works once but then does not stop writing when other gestures are activated. Marker continues to draw until I use the erase gesture which erases everything but also completely prevents the marker from working again until I reload the avatar.

**Comment by wickles** [comment]
I figured out the problem and have a solution but it's bothering me that I don't really understand what's happening still. 

The problem in my case is that the EraseAll animation sets `Trail Renderer.Time` to 0 and it gets stuck at that value permanently. It doesn't seem to matter if I have Write Defaults on or off on any of the marker states. It seems like the marker is designed under the assumption that `Trail Renderer.Time` will reset back to its original value somehow (Write Defaults?) but it is never explicitly changed anywhere except on the EraseAll animation. So my solution is to explicitly set the `Time` value back to 120 in my MarkerOn animation.

**Comment by chibiskuld** [comment]
Sounds like write defaults was disabled.

**Comment by wickles** [comment]
> It doesn't seem to matter if I have Write Defaults on or off

**Comment by wickles** [comment]
To be clear it does not work with Write Defaults on for the marker states like you are suggesting it should. I guess there is another part of the animation controller interfering somehow. The animation controller is modified from an older version that used one of the old networked int toggle systems, so I can see how there might be something odd going on there. 

I do not think it is a good idea to rely on the `Time` value getting reset by some arcane Write Defaults invocation. May be due for a PR. 

Related https://github.com/theepicsnail/Marker/issues/11

**Comment by camillaMay** [comment]
Hello, and thank you for the awesome marker :D
I have the same issue. Market never stops writing if I leave any write defaults on.
I found this workaround to be a working solution for me : 
https://translate.google.com/translate?sl=auto&tl=en&u=https://www.holynight.net/archives/4366

I hope this can be of any help !
